Kim staging

klazarsk opened the pull request:
Initial PR to incorporate alert_smtp_filtered.sh and alert_syslog.sh to enhance the sample alert agents bundled with Pacemaker. These additional sample alert agents with filtering enhancements were created pursuant to customer requests. Also included man pages for each, with the man page source in markdown format for easy maintainability.

klazarsk committed:
Initial checkin on rebased repo.

klazarsk committed:
placeholder README

klazarsk committed:
Ansible updated to reflect filename change, initial alert_smtp_filtered man page created

klazarsk committed:
"attribute" section was mis-named "resource"

Incomplete sentence before the second example completed

The bad command with the mis-named agent has been removed:

[root@nodea ~]# pcs alert recipient add smtp-alert value=student@workstation.lab.example.com
Error: alert 'smtp-alert' does not exist

klazarsk committed:
modified:   alert_smtp_filtered.5.md

INSTALLATION section added
Formatting improved
Typo fixed
Examples descriptions improved

modified:   alert_smtp_filtered.5

Man page regenerated to incorporate changes from markdown source file

new file:   alert_syslog.5.md

Markdown source file for alert_syslog.sh man page added

new file:   alert_syslog.5

The generated alert_syslog.sh man page

wenningerk commented on agents/alert_smtp_filtered.sh:
```diff
@@ -0,0 +1,159 @@
+#!/bin/sh
+#
```
Starting with the copyright header is good practice in many projects like pacemaker where this is derived from. Maybe we should stick to that.

wenningerk commented on agents/alert_smtp_filtered.sh:
```diff
@@ -0,0 +1,159 @@
+#!/bin/sh
+#
+# Adapted from alert_smtp.sh from the pacemaker package by 
```
Why do we want to go for 2 versions of that alert-agent?
Can't we do the filtering as that it is deactivated by default?
I read you wanted to keep the original in its simple shape to reduce risk of errors but shouldn't we be able to separate the code so that disabling the filtering becomes simple enough?
Then we don't have to explain what it is derived from as the git-history has it all.

klazarsk replied:
Understood and agree - I'll make that change from my kim-staging branch and re-submit.

wenningerk replied:
And sharing as well raises chances issues are found.

wenningerk commented on agents/README.md:
```diff
@@ -1 +1,249 @@
 # alert-agents
+
+# alert_syslog.sh: Sample Pacemaker Syslog Alert Agent 
+
```
Now that we are packaging alert-agent independently we should go for stuff that is useful as is (which it is of course but we might be bold enough to tell instead of calling it a sample in the first line ;-) )

wenningerk replied:
If we want to have a readme that detailed we should probably have one file per agent or something.
If we want to have man-pages on top we could think of trying to have just a single readme with minimum information that isn't gonna change probably. The alert-agents we have so far don't create meta-data and don't have a usage output. But as they are shell-executables we might add a usage-output and try to
dynamically create man-pages from that. So implementation and documentation would be in the same file. One can have some help on the cmdline. And everything doesn't get out of sync.

wenningerk commented on ansible/debug_alert_smtp.yml:
```diff
@@ -0,0 +1,23 @@
+---
```
Now that we have a repo for the alert-agents shouldn't we simply package them?
No non-developer-installation (and even there I try to prevent that where possible) should have executables that are not in the rpm-database.

klazarsk replied:
mmm, My thinking was to give sample ansible for deployment, but of course your point is valid, and after considering it, I'm realizing that if they're skilled enough to modify and debug the agents, they're likely more than capable of deploying them.

wenningerk replied:
Giving an example how one could do quick deployment during while working on an alert-agent for instance is not a bad idea. Guess I would like to offer it as this instead of making it look as if this would be the preferred way of deployment for a stable alert-agent.

wenningerk commented on agents/alert_syslog_debug.sh:
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Idea of a separate repo is that we can package the stuff in here.
This as well means that we don't want to copy executables around.
Can't we just control debug via configuration.

klazarsk replied:
That's fair!

wenningerk reviewed (commented):
Review isn't actually a real review of the new code yet but I've added a few ideas where they came up reading through the PR. It is late already on my side of the world and I wanted to make that visible so I'm still submitting this review.

klazarsk committed:
modified:   agents/README.md
	modified:   agents/alert_smtp_filtered.sh
	modified:   agents/alert_syslog.sh
	modified:   docs/alert_smtp_filtered.5
	modified:   docs/alert_smtp_filtered.5.md
	modified:   docs/alert_syslog.5
	modified:   docs/alert_syslog.5.md

Removed reference to "attributes" alerts as they are not a thing. Per
documentation, supported alerts are "node,fencing,resource"

Re-generated the man pages to reflect the .md source updates

klazarsk committed:
modified:   alert_smtp.sh

Removed "attribute" filter type, as valid alert kinds are
fencing,node,resource per documentation.

Slight formatting changes

Added debug trace options, commented out by default, which
redirect trace and stderr messages to /var/log/pacemaker-alert_smtp-debug.log

```
\# set -xv
\# exec 2 > /var/log/pacemaker-alert_smtp-debug.log
```

modified:   alert_syslog.sh

Removed "attribute" filter type, as valid alert kinds are
fencing,node,resource per documentation.

Added debug trace options, commented out by default, which
redirect trace and stderr messages to /var/log/pacemaker-alert_syslog-debug.log

```
\# set -xv
\# exec 2 > /var/log/pacemaker-alert_smtp-debug.log
```
deleted:    alert_smtp_filtered.sh

Moved alert_smtp_filtered.sh to alert_smtp.sh

deleted:    alert_smtp_filtered_debug.sh
deleted:    alert_syslog_debug.sh

Removed the wrapper script in favor of turning on tracing within the
script, commented out by default.

klazarsk committed:
Users advanced enough to customize alert agents, should be advanced enough to write ansible.

klazarsk committed:
Starting on putting help page into agent

klazarsk committed:
README.md

Slimmed down to list of alert agent, feature changes, and barebones list
of options for each.

Pointer to appropriate man page and help screens for each (man pages and
help screens pending, in progress)

klazarsk committed:
renamed man source and rendered files to reflect agent name change

klazarsk committed:
modified:   alert_smtp.sh

Added help screen

modified:   ../docs/alert_smtp.5.md

Changed reference to previous filename to reflect the enhanced
agent replacing alert_smtp.

modified:   ../docs/alert_smtp.5

Re-rendered alert_smtp.5 to incorporate changes to markdown
file.

klazarsk committed:
modified:   agents/alert_smtp.sh
modified:   agents/alert_syslog.sh

Completed help screens

modified:   docs/alert_syslog.5.md

Improved some  verbiage and formatting

modified:   docs/alert_syslog.5

Regenerated to incorporate changes in the markdown source file

klazarsk committed:
modified:   docs/alert_syslog.5.md

Added trailing \ to partial command line in example

klazarsk committed:
modified:   docs/alert_smtp.5.md

Minor verbiage improvement

klazarsk committed:
modified:   alert_smtp.5
modified:   alert_syslog.5

Regenerated to included the latest minor verbiage tweaks.

klazarsk committed:
Edited man page .md files to comply with formatting conventions, WIP testing calling man from alert_smtp.sh. Will complete alert_smtp.sh tomorrow.

klazarsk committed:
modified:   agents/alert_smtp.sh

Moved documentation to man page, script when executed in interactive
session now calls up the man page.

modified:   agents/alert_syslog.sh

Moved documentation to man page, script when executed in interactive

modified:   docs/alert_smtp.5
modified:   docs/alert_smtp.5.md

Fixed various errors in the documentation, fixed errant references
to alert_syslog (which was authored first, result of cp and not
catching required updates)
